index.js

```diff
@@ -13,352 +13,397 @@
+      throw new Error('ember-cli-amd: You must specify a loader option the amd options in ember-cli-build.js.');
     }
-    if (useRequire || useDojo) {
-      amdBuilder(names);
+
+    if ((app.options.amd.loader === 'requirejs' || app.options.amd.loader === 'dojo') && !app.options.amd.libraryPath) {
+      throw new Error('ember-cli-amd: When using a local loader, you must specify its location with the amdBase property in the amd options in ember-cli-build.js.');
     }
-    modules = unique;
   },
-  postprocessTree: function(type, tree) {
+
+  postprocessTree: function (type, tree) {
+
+    if (type !== 'all')
+      return tree;
+
+    var outputPaths = this.app.options.outputPaths;
+    
+    // Create the string replace patterns for the various application files
+    // We will replace require and define function call by their pig-latin version
     var data = {
       files: [
-        new RegExp(appName + '(.*js)'),
-        new RegExp('vendor(.*js)'),
-        'assets/test-support.js'
+        new RegExp(path.parse(outputPaths.app.js).name + '(.*js)'),
+        new RegExp(path.parse(outputPaths.vendor.js).name + '(.*js)'),
+        new RegExp(path.parse(outputPaths.tests.js).name + '(.*js)'),
+        new RegExp(path.parse(outputPaths.testSupport.js.testSupport).name + '(.*js)')
       ],
       patterns: [
         { match: /(\W|^|["])define(\W|["]|$)/g, replacement: '$1efineday$2' },
         { match: /(\W|^|["])require(\W|["]|$)/g, replacement: '$1equireray$2' }
       ]
     };
+    var dataTree = stringReplace(tree, data);
+
+    // Special case for the test loader that is doing some funky stuff with require
     var testLoader = {
       files: [
-        'assets/test-loader.js'
+        new RegExp(path.parse(outputPaths.testSupport.js.testLoader).name + '(.*js)')
       ],
       patterns: [
         { match: /(\W|^|["])define(\W|["]|$)/g, replacement: '$1efineday$2' },
         { match: /[^.]require([(])/g, replacement: 'equireray(' }
       ]
     };
-    var dataTree = replace(tree, data);
-    return replace(dataTree, testLoader);
+
+    return stringReplace(dataTree, testLoader);
   },
-  preBuild: function() {
-    if (isBuilt) {
-      var data = findAMD();
-      if (data.names !== names) {
-        names = data.names;
-        if (useDojo) {
-          names = "'dojo/dojo'," + names;
-        }
-        if (useRequire || useDojo) {
-          amdBuilder(names);
-        }
-        isBuilt = true;
-        modules = data.modules;
-      }
+
+  postBuild: function (result) {
+    // When ember build --watch or ember serve are used, this function will be called over and over 
+    // as a user updates code. We need to figure what we have to build or copy.
+  
+    // If it is using a file for the amd configuration, we will just copy it. We don't need to 
+    // use sha to define if we need to copy it or not, the price is equivalent.
+    if (typeof this.app.options.amd.config === 'string') {
```

Comment:
Should add a check if `this.options.amd` exists too and fail gracefully or it will error out if no `amd` property has been set yet crashing the build.